core/dbt/cli/params.py

```diff
@@ -1,166 +1,226 @@
     "--defer/--no-defer",
+    envvar="DBT_DEFER",
     help="If set, defer to the state variable for resolving unselected nodes.",
-    default=True,
+)
+
+enable_legacy_logger = click.option(
+    "--enable-legacy-logger/--no-enable-legacy-logger",
+    envvar="DBT_ENABLE_LEGACY_LOGGER",
+    hidden=True,
 )
 
 event_buffer_size = click.option(
     "--event-buffer-size",
+    envvar="DBT_EVENT_BUFFER_SIZE",
     help="Sets the max number of events to buffer in EVENT_HISTORY.",
     default=100000,
     type=click.INT,
 )
 
-exclude = click.option("--exclude", help="Specify the nodes to exclude.")
+exclude = click.option("--exclude", envvar=None, help="Specify the nodes to exclude.")
 
 fail_fast = click.option(
-    "--fail-fast/--no-fail-fast", "-x/ ", help="Stop execution on first failure.", default=False
+    "--fail-fast/--no-fail-fast",
+    "-x/ ",
+    envvar="DBT_FAIL_FAST",
+    help="Stop execution on first failure.",
 )
 
 full_refresh = click.option(
     "--full-refresh",
+    envvar="DBT_FULL_REFRESH",
     help="If specified, dbt will drop incremental models and fully-recalculate the incremental table from the model definition.",
     is_flag=True,
 )
 
 indirect_selection = click.option(
-    "--indirect_selection",
+    "--indirect-selection",
+    envvar="DBT_INDIRECT_SELECTION",
     help="Select all tests that are adjacent to selected resources, even if they those resources have been explicitly selected.",
     type=click.Choice(["eager", "cautious"], case_sensitive=False),
     default="eager",
 )
 
+#  TODO: Should this be exposed in the help text?
+log_cache_events = click.option(
+    "--log-cache-events/--no-log-cache-events",
+    envvar="DBT_LOG_CACHE_EVENTS",
+    hidden=True,
+)
+
 log_format = click.option(
     "--log-format",
+    envvar="DBT_LOG_FORMAT",
     help="Specify the log format, overriding the command's default.",
     type=click.Choice(["text", "json", "default"], case_sensitive=False),
     default="default",
 )
 
 log_path = click.option(
     "--log-path",
+    envvar="DBT_LOG_PATH",
     help="Configure the 'log-path'. Only applies this setting for the current run. Overrides the 'DBT_LOG_PATH' if it is set.",
     type=click.Path(),
 )
 
-models = click.option("-m", "-s", help="Specify the nodes to include.", multiple=True)
+macro_debugging = click.option(
+    "--macro-debugging/--no-macro-debugging",
+    envvar="DBT_MACRO_DEBUGGING",
+    hidden=True,
+)
+
+models = click.option(
+    "-m",
+    "-s",
+    "models",
+    envvar=None,
+    help="Specify the nodes to include.",
+    multiple=True,
+)
 
 output = click.option(
     "--output",
+    envvar=None,
     help="TODO: No current help text",
     type=click.Choice(["json", "name", "path", "selector"], case_sensitive=False),
     default="name",
 )
 
-ouptut_keys = click.option(
+output_keys = click.option(
     "--output-keys",
+    envvar=None,
     help="TODO: No current help text",
-    default=False,
+    is_flag=True,
```

Comment:
It looks like other places that use `is_flag` are boolean flags. If this is accurate, I think `--output-keys` is a comma separated string of values, so I'm not sure if this should be using `is_flag`.